go/integTest/advanced_commands_test.go

```diff
@@ -0,0 +1,546 @@
+		}
+		hosts = append(hosts, api.NodeAddress{Host: parts[0], Port: port})
+	}
+	return hosts
+}
+
+func (suite *AdvancedTestSuite) SetupSuite() {
+	// Stop any existing clusters
+	suite.runClusterManager([]string{"stop", "--prefix", "cluster"}, true)
+
+	// Start a standalone instance if no endpoints are provided
+	if *advancedStandaloneHosts == "" {
+		clusterManagerOutput := suite.runClusterManager([]string{"start", "-r", "3"}, false)
+		suite.standaloneHosts = suite.extractAddresses(clusterManagerOutput)
+	} else {
+		suite.standaloneHosts = suite.parseHosts(*advancedStandaloneHosts)
+	}
+
+	// Start a cluster instance
+	clusterManagerOutput := suite.runClusterManager([]string{"start", "--cluster-mode"}, false)
+	suite.clusterHosts = suite.extractAddresses(clusterManagerOutput)
+
+	suite.tls = *advancedTls
+
+	// Get server version
+	suite.serverVersion = suite.getServerVersion()
+}
+
+func (suite *AdvancedTestSuite) getServerVersion() string {
+	client := suite.defaultClient().(api.GlideClientCommands)
+	defer client.Close()
+
+	info, err := client.Info()
+	assert.Nil(suite.T(), err)
+
+	// Extract version from INFO output
+	lines := strings.Split(info, "\n")
+	for _, line := range lines {
+		if strings.HasPrefix(line, "redis_version:") {
+			return strings.TrimSpace(strings.TrimPrefix(line, "redis_version:"))
+		}
+	}
+	return ""
+}
+
+func (suite *AdvancedTestSuite) TearDownSuite() {
+	// Stop the cluster when done
+	suite.runClusterManager([]string{"stop", "--prefix", "cluster"}, true)
+}
+
+func (suite *AdvancedTestSuite) SetupTest() {
+	// Initialize client using the default configuration
+	suite.client = suite.defaultClient()
+}
+
+func (suite *AdvancedTestSuite) defaultClientConfig() *api.GlideClientConfiguration {
+	return api.NewGlideClientConfiguration().
+		WithAddress(&suite.standaloneHosts[0]).
+		WithUseTLS(suite.tls).
+		WithRequestTimeout(5000)
+}
+
+func (suite *AdvancedTestSuite) defaultClient() api.BaseClient {
+	config := suite.defaultClientConfig()
+	client, err := api.NewGlideClient(config)
+	assert.Nil(suite.T(), err)
+	assert.NotNil(suite.T(), client)
+	return client
+}
+
+func (suite *AdvancedTestSuite) defaultClusterClientConfig() *api.GlideClusterClientConfiguration {
+	return api.NewGlideClusterClientConfiguration().
+		WithAddress(&suite.clusterHosts[0]).
+		WithUseTLS(suite.tls).
+		WithRequestTimeout(5000)
+}
+
+func (suite *AdvancedTestSuite) defaultClusterClient() api.GlideClusterClientCommands {
+	config := suite.defaultClusterClientConfig()
+	return suite.clusterClient(config)
+}
+
+func (suite *AdvancedTestSuite) clusterClient(config *api.GlideClusterClientConfiguration) api.GlideClusterClientCommands {
+	client, err := api.NewGlideClusterClient(config)
+	assert.Nil(suite.T(), err)
+	assert.NotNil(suite.T(), client)
+	suite.clusterClients = append(suite.clusterClients, client)
+	return client
+}
+
+func (suite *AdvancedTestSuite) TestFunctionKillNoWrite() {
```

Comment:
There is significant code duplication between the various tests. Create helper functions to consolidate code and make the actual logic being tested easier to identify and validate. The go func() calls specifically look almost identical with small variations that could be handled with parameterization.